sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -497,7 +497,7 @@ private void ensureWindowsAreASet() {
       // right, and cast the window type away here.
       @SuppressWarnings({"unchecked", "rawtypes"})
       Coder<Collection<? extends BoundedWindow>> collectionCoder =
-          (Coder) CollectionCoder.of(this.windowCoder);
```

Comment:
Will this break DataflowRunner's update by changing all these coders?

Reply:
The encoded/decoded bytes will still be the same since both ListCoder and CollectionCoder rely on the IterableLikeCoder for encoding/decoding records.

Does Dataflows update care about the names of the coders used or only about the encoding?

Reply:
I believe Dataflow looks at the coder id to determine compatibility (if the coder id has changed, it assumes the update is incompatible).